### DISCUSSION TITLE
How to handle the input and output file path in process exec block?

### DISCUSSION THREAD

**Discussion by wangzc66** [nextflow-io/nextflow#5407] [Q&A]
In Nextflow, exec blocks are a special construct that allows you to run a script directly in the process using Groovy code, but unlike the script block, it does not automatically stage files into the process execution directory.
Is it better to avoid using file paths in the exec block?

**Comment by bentsherman** [comment]
There is an old issue about this but I can't find it. The `exec` block doesn't work correctly with `path` inputs/outputs, so you should use `val` inputs/outputs instead. Unfortunately you lose some provenance information when you do that, so it's something I'd like to fix in the future
